test/realm_test.dart

```diff
@@ -194,6 +194,112 @@ Future<void> main([List<String>? args]) async {
       config.schemaVersion = 3;
       expect(config.schemaVersion, equals(3));
     });
+
+    test('Configuration isReadOnly property - realm files must exist', () {
+      Configuration config = Configuration([Car.schema]);
+      config.isReadOnly = true;
+      //You cannot open Realm in read-only mode if Realm files do not exist.
+      //Files are created the first time Realm is opened.
+      //Opening Realm in read-only mode can be done after reopening existing Realm with read-only configuration.
+      //First opening of Realm if configuration is read-only will throw an exception.
+      expect(() => Realm(config), throws<RealmException>("Message: No such table exists"));
+    });
+
+    test('Configuration readOnly argument - realm files must exist', () {
+      Configuration config = Configuration([Car.schema], readOnly: true);
+      expect(() => Realm(config), throws<RealmException>("Message: No such table exists"));
+    });
+
+    test('Configuration read-only - open existing realm with read-only config', () {
+      //Create realm and its files
+      Configuration config = Configuration([Car.schema]);
+      var realm = Realm(config);
+      realm.close();
+
+      //Reconfigure existing realm for read-only mode.
+      config = Configuration([Car.schema], readOnly: true);
+      realm = Realm(config);
+      realm.close();
+    });
+
+    test('Configuration read-only - reading is possible', () {
+      Configuration config = Configuration([Car.schema]);
+      var realm = Realm(config);
+      realm.write(() => realm.add(Car("Mustang")));
+      realm.close();
+
+      //Reconfigure existing realm for read-only mode.
+      config.isReadOnly = true;
+      realm = Realm(config);
+      var cars = realm.all<Car>();
+      realm.close();
+    });
+
+    test('Configuration read-only - writing on read-only Realms throws', () {
+      Configuration config = Configuration([Car.schema]);
+      var realm = Realm(config);
+      realm.close();
+
+      //Reconfigure existing realm for read-only mode.
+      config = Configuration([Car.schema], readOnly: true);
+      realm = Realm(config);
+      expect(() => realm.write(() => realm.add(Car("Mustang"))), throws<RealmException>("Can't perform transactions on read-only Realms."));
+      realm.close();
+    });
+
+    test('Configuration read-only - editing realm objects throws', () {
+      Configuration config = Configuration([Car.schema]);
+      var realm = Realm(config);
+      var car = Car("Mustang");
+      realm.write(() => realm.add(car));
+      realm.close();
+
+      config.isReadOnly = true;
+      realm = Realm(config);
+      expect(() => realm.write(() => car.make = "Opel"), throws<RealmException>("Can't perform transactions on read-only Realms."));
+      realm.close();
+    });
+
+    test('Configuration read-only - deleting realm object throws', () {
```

Comment:
This tests is over doing it. I suggest we delete it. We have a test above that does test that `write` call fails.

Reply:
OK